wwwbase/js/scramble.js

```diff
@@ -0,0 +1,1051 @@
+$(document).ready(function() {
+   
+  function callAjax(level)
+  {
+    var result;  
+     $.ajax({
+        type: "POST",
+        url: wwwRoot + "ajax/scramble.php",
+        async: false,
+        data: 'difficulty='+level,
+        datatype: "html",
+      })
+      .done(function(response) {      
+        result = $.parseJSON(response);
+        //console.log(result);
+      })
+      .fail(function() {
+        console.log("Nu merge");
+      });
+    console.log(result);
+    return result;
+  }
+
+  function  reposArray(array, pos1, pos2) 
+  {
+    var i, tmp;
+    //cast inputs as integers
+    pos1 = parseInt(pos1, 10);
+    pos2 = parseInt(pos2, 10);    
+    // if positions are different inside array
+    if (pos1 !== pos2 && 0 <= pos1 && pos1 <= array.length && 0 <= pos2 && pos2 <= array.length) 
+      {
+      //save element from pos1
+      tmp = array[pos1];
+      // move element down and shift other elements up
+      if(pos1 < pos2)
+      {
+        for(i = pos1; i < pos2; i++)
+        {
+          array[i] = array[i+1];
+        }
+      }
+      else // move element up and shift other elements down
+      {
+        for(i = pos1; i > pos2; i--)
+        {
+          array[i] = array [i - 1];
+        }
+      }
+      //put element from position 1 to destination
+      array[pos2] = tmp;          
+    }
+     return array;
+  }
+
+  var score = 0;
+  var cnt = 0;
+  var searchWord;
+  var lettersPressed = new String(); // in acest array se retin literele tastate
+  var totalWords = new Array(); // the possible words that can be made from the randomWord.
+  var difficulty; // initial selected difficulty
+  var layers = []; // Array of currently drawn layers.
+  var upLayers = []; // Letters in the top 
+  var downLayers = []; // Letters in the bottom
+  var threshold = 120; // y axis limit for moving letters
+  var layerSpeed = 200; //Global animation speed
+
+
+
+  function selectDifficulty() {
+
+    $(".difficultyButton").on("click", function() {
+      // this e pentru a prelua valoarea butonului tocmai apasat, si nu a unuia oarecare
+      difficulty = $(this).attr("value");
+      $(this).blur();
+        
+      var word = callAjax(difficulty);
+      if(word.everyWord.length < 4) {
+        word = callAjax(difficulty);
+      }
+      cnt = 0;
+      lettersPressed = [];
+      totalWords = word.everyWord;
+
+      for(var i = 0; i < totalWords.length; i++)
+      {
+        totalWords[i] = totalWords[i].toLowerCase();
```

Comment:
Le preiau, le fac to lower case, si le caut in array-ul primit prin ajax. Diacriticile apar si in literele desenate asa ca am decis sa nu scot diacriticile.